routing/transit_graph.cpp

```diff
@@ -183,25 +340,49 @@ void TransitGraph::Fill(transit::GraphData const & transitData, GateEndings cons
 
 bool TransitGraph::IsGate(Segment const & segment) const
 {
-  return m_segmentToGate.count(segment) > 0;
+  if (m_transitVersion == ::transit::TransitVersion::OnlySubway)
+    return m_segmentToGateSubway.count(segment) > 0;
+  else if (m_transitVersion == ::transit::TransitVersion::AllPublicTransport)
+    return m_segmentToGatePT.count(segment) > 0;
+  CHECK(false, (m_transitVersion));
 }
 
 bool TransitGraph::IsEdge(Segment const & segment) const
 {
-  return m_segmentToEdge.count(segment) > 0;
+  if (m_transitVersion == ::transit::TransitVersion::OnlySubway)
+    return m_segmentToEdgeSubway.count(segment) > 0;
+  else if (m_transitVersion == ::transit::TransitVersion::AllPublicTransport)
+    return m_segmentToEdgePT.count(segment) > 0;
+  CHECK(false, (m_transitVersion));
 }
 
 transit::Gate const & TransitGraph::GetGate(Segment const & segment) const
 {
-  auto const it = m_segmentToGate.find(segment);
-  CHECK(it != m_segmentToGate.cend(), ("Unknown transit segment."));
+  auto const it = m_segmentToGateSubway.find(segment);
+  CHECK(it != m_segmentToGateSubway.cend(), ("Unknown transit segment."));
```

Comment:
`CHECK(it != m_segmentToGateSubway.cend(), ("Unknown transit segment:", segment));`?